client/option_advanced.go

```diff
@@ -206,6 +206,23 @@ func WithBoundHandler(h remote.BoundHandler) Option {
 	return Option{F: func(o *client.Options, di *utils.Slice) {
 		di.Push(fmt.Sprintf("AddBoundHandler(%T)", h))
 
-		o.RemoteOpt.AppendBoundHandler(h)
+		existInbound := false
+		existOutbound := false
+		for _, inboundHandler := range o.RemoteOpt.Inbounds {
+			if inboundHandler == h.(remote.InboundHandler) {
```

Comment:
It seems you didn't judge whether it's ok to assert h to remote.InboundHandler?
Actually Don't need to check both Inbounds and Outbounds, if h implements remote.InboundHandler, check whether there's a same one, and the other is the same.

Reply:
@Marina-Sakai 
I think so. You may check if h is InboundHandler or OutboundHandler at first.
There is another case that the h is DuplexBoundHandler. If so, it should be in both Inbounds and Outbounds. Check one of them should be fine.

Reply:
@jayantxie @ppzqh Fixed. Thank you for your comments! Could you review it again?